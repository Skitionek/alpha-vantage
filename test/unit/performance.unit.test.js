
import AlphaVantageMock from "../mocks/alphaVantageMock";
import { matchesSnapshot } from "../jest.extensions";

let alpha;
beforeAll(() => {
	alpha = new AlphaVantageMock();
});

test(`sector performance data works`, () =>
	alpha.performance.sector()
		.then(matchesSnapshot)
);
