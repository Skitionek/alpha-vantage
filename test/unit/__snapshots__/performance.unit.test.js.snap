// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sector performance data works 1`] = `
Object {
  "information": null,
  "last_refreshed": null,
  "rank_a_real_time_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "real_estate": null,
    "utilities": null,
  },
  "rank_b_day_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "real_estate": null,
    "utilities": null,
  },
  "rank_c_day_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "real_estate": null,
    "utilities": null,
  },
  "rank_d_month_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "real_estate": null,
    "utilities": null,
  },
  "rank_e_month_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "real_estate": null,
    "utilities": null,
  },
  "rank_f_year_to_date_ytd_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "real_estate": null,
    "utilities": null,
  },
  "rank_g_year_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "real_estate": null,
    "utilities": null,
  },
  "rank_h_year_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "real_estate": null,
    "utilities": null,
  },
  "rank_i_year_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "utilities": null,
  },
  "rank_j_year_performance": Object {
    "communication_services": null,
    "consumer_discretionary": null,
    "consumer_staples": null,
    "energy": null,
    "financials": null,
    "health_care": null,
    "industrials": null,
    "information_technology": null,
    "materials": null,
    "utilities": null,
  },
}
`;
